include/fmt/os.h

```diff
@@ -379,56 +379,76 @@ struct ostream_params {
 #  endif
 };
 
-FMT_END_DETAIL_NAMESPACE
-
-// Added {} below to work around default constructor error known to
-// occur in Xcode versions 7.2.1 and 8.2.1.
-constexpr detail::buffer_size buffer_size{};
-
-/** A fast output stream which is not thread-safe. */
-class FMT_API ostream final : private detail::buffer<char> {
- private:
+template <typename Char> class file_buffer final : public detail::buffer<Char> {
```

Comment:
This doesn't need to be a template because file I/O is char-based.

Reply:
> doesn't need to be

That is exactly what I've tried in the [previous commit](https://github.com/fmtlib/fmt/compare/master...e75f9b8e38c9ca63f94e4cf9b4524ee89c14cb83).
But compilers generally are [not happy](https://github.com/fmtlib/fmt/actions/runs/3248601160/jobs/5330007655) about it.

I simply took the same approach as with `memory_buffer`.

Reply:
You should be able to fix that warning by outlining some virtual method. `memory_buffer` is different because it has to be a template.

Reply:
> by outlining some virtual method

That I [did](https://github.com/fmtlib/fmt/pull/3139/commits/ef4074802a2b528d3293f4e1e3ac6da0deeece20#diff-bae4f42f207eec1b8dba3ed186368b3b1a83803e33b9a47f56854eb211be0d23R385) and got a [different problem](https://github.com/fmtlib/fmt/actions/runs/3245229029/jobs/5322483555) - the v-table gets *anchored* into a shared library but *not exported*, which makes it unavailable outside of a shared library.

Initially I thought that the only fix would be to export the entire class (to export a v-table) which is the opposite of this entire change. But after giving it a second though - I think it would be enough to outline c-tor and other special functions. Let me try that.

Reply:
done - it worked